gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -159,7 +164,7 @@ private void showAddFeatureDialog(Pair<ImmutableList<Layer>, Point> args) {
     ImmutableList<Layer> layers = args.first;
     Point point = args.second;
     addFeatureDialogFragment.show(
-        layers, getChildFragmentManager(), layer -> viewModel.addFeature(layer, point));
+        layers, getChildFragmentManager(), layer -> showFeatureTypeDialog(layer, point));
```

Comment:
We need to check whether the user is allowed to add both points and polygons to decide whether or not they should be shown the feature type selector dialog.

In order to know this, we need to know which layer they want to add a feature to, and therefore we must show the layer selector first.

If the user is the owner or manager of the project, they can add points and polygons to all layers. If they are a contributor, they should only be allowed to add what's specified in layer.getContributorsCanAdd().

Reply:
added check.